### PR TITLE
Update

### LINKED ISSUE #15: [feat] change api for widget
```js
const widget = require('cache-element/widget')

module.exports = createWidget

function createWidget () {
  // variables are bound in the closure
  return widget({
    onload: function () {
    },
    onunload: function () {
    },
    onupdate: function (link) {
    },
    render: function () {
      // return html
    }
  })
}
```
Closer to react's stateful components but without all the `this` stuff - also heaps less confusing and relying on implicit ordering of things

### PR DESCRIPTION
**Pull request by yoshuawuyts**
update elements. Closes #15 #16 - makes this package a bit more wieldy.

cc/ @emmak would this API work well for you?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,83 +16,85 @@ React's `.shouldComponentUpdate()` method, but only using native DOM methods.
...
-    <p>The person's name is ${name}</p>
-  `
+var cache = require('cache-element')
+var html = require('bel')
+
+var element = Element()
+
+let el = element('Tubi') // creates new element
+let el = element('Tubi') // returns cached element (proxy)
+let el = element('Babs') // creates new element
```
Comment (marlun) [inline README.md]: Are you missing age parameter in these?

**Comment by yoshuawuyts** [comment]
updated

**Comment by timwis** [comment]
+1 on making the API a bit easier to use.

1. The times I've used `widget`, I've always had to store references outside my `onupdate` function ([example](https://github.com/timwis/choo-leaflet-demo/blob/master/src/map.js)). Now that `onupdate` and `render` are separate, will it still be possible to do that? (quite necessary for most things I think)

2. This is a breaking change to the API so I assume this will be a major version update, right?

3. With a major change like this, my thought goes to whether it will still work with things like leaflet, d3, codemirror, etc. -- I wonder if there's a way to verify that via tests?

**Comment by yoshuawuyts** [comment]
`v2.0.0`

**Pull request merged**